kratos/tests/cpp_tests/elements/test_levelset_convection_element_simplex.cpp

```diff
@@ -0,0 +1,277 @@
+        model_part.GetProcessInfo().SetValue(DYNAMIC_TAU, 0.001);
+        model_part.GetProcessInfo().SetValue(DELTA_TIME, delta_time);
+
+        ConvectionDiffusionSettings::Pointer p_conv_diff_settings = Kratos::make_unique<ConvectionDiffusionSettings>();
+        model_part.GetProcessInfo().SetValue(CONVECTION_DIFFUSION_SETTINGS, p_conv_diff_settings);
+        model_part.GetProcessInfo().SetValue(CROSS_WIND_STABILIZATION_FACTOR, cross_wind_diff);
+        p_conv_diff_settings->SetUnknownVariable(DISTANCE);
+        p_conv_diff_settings->SetConvectionVariable(VELOCITY);
+
+        // Set the element properties
+        Properties::Pointer pElemProp = model_part.CreateNewProperties(0);
+
+        // Geometry creation
+        model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+        model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+        model_part.CreateNewNode(3, 0.0, 1.0, 0.0);
+        model_part.CreateNewNode(4, 0.0, 0.0, 1.0);
+        std::vector<ModelPart::IndexType> elemNodes {1, 2, 3, 4};
+        auto p_element = model_part.CreateNewElement("LevelSetConvectionElementSimplex3D4N", 1, elemNodes, pElemProp);
+        
+        // Define the nodal values
+        Matrix vel_original(4,3);
+        vel_original(0,0) = 0.0; vel_original(0,1) = 0.1; vel_original(0,2) = 0.1;
+        vel_original(1,0) = 0.1; vel_original(1,1) = 0.2; vel_original(1,2) = 0.2;
+        vel_original(2,0) = 0.2; vel_original(2,1) = 0.3; vel_original(2,2) = 0.3;
+        vel_original(3,0) = 0.2; vel_original(3,1) = 0.3; vel_original(3,2) = 0.3;
+
+        for(unsigned int i=0; i<4; i++){
+            for(unsigned int k=0; k<3; k++){
+                p_element->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY)[k]    = vel_original(i,k);
+                p_element->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY, 1)[k] = 0.9*vel_original(i,k);
+                p_element->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY, 2)[k] = 0.75*vel_original(i,k);
+            }
+        }
+        p_element->GetGeometry()[0].FastGetSolutionStepValue(DISTANCE) = -1.0;
+        p_element->GetGeometry()[1].FastGetSolutionStepValue(DISTANCE) = -1.0;
+        p_element->GetGeometry()[2].FastGetSolutionStepValue(DISTANCE) =  0.5;
+        p_element->GetGeometry()[3].FastGetSolutionStepValue(DISTANCE) =  0.5;
+
+        p_element->GetGeometry()[0].FastGetSolutionStepValue(DISTANCE,1) = -0.8;
+        p_element->GetGeometry()[1].FastGetSolutionStepValue(DISTANCE,1) = -0.8;
+        p_element->GetGeometry()[2].FastGetSolutionStepValue(DISTANCE,1) =  0.7;
+        p_element->GetGeometry()[3].FastGetSolutionStepValue(DISTANCE,1) =  0.7;
+
+        // Compute RHS and LHS
+        Vector rhs = ZeroVector(5);
+        Matrix lhs = ZeroMatrix(5,4);
+
+        p_element->Initialize();
+        const ProcessInfo& const_process_info = model_part.GetProcessInfo();
+        p_element->CalculateLocalSystem(lhs, rhs, const_process_info);
+
+        KRATOS_CHECK_EQUAL(rhs.size(), 4);
```

Comment:
please use the matrix comparison check macros

Reply:
right, I forgot about those... done!